src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -33,40 +33,47 @@
     (.clear engine)
     (.add engine (clj->js datums))))
 
+
 (react/defc- MethodDetailsViewer
   {:get-fields
    (fn [{:keys [refs]}]
      ((@refs "methodDetails") :get-fields))
+   :clear-redacted-snapshot
+   (fn [{:keys [refs]}]
+     ((@refs "methodDetails") :clear-redacted-snapshot))
+   :get-initial-state
+   (fn [{:keys [props]}]
+     {:redacted? (:redacted? props)})
    :render
    (fn [{:keys [props state]}]
-     (let [{:keys [loaded-method error]} @state]
+     (let [{:keys [loaded-method error redacted?]} @state]
        (cond loaded-method [comps/EntityDetails
                             (merge {:ref "methodDetails"
-                                    :snapshots (get (:methods props)
-                                                    (replace loaded-method [:namespace :name]))
-                                    :entity loaded-method}
-                                   (select-keys props [:editing? :wdl-parse-error :onSnapshotIdChange]))]
+                                    :entity loaded-method
+                                    :redacted? redacted?}
+                                   (select-keys props [:editing? :snapshots :wdl-parse-error :onSnapshotIdChange]))]
              error (style/create-server-error-message error)
              :else [comps/Spinner {:text "Loading details..."}])))
    :component-did-mount
    (fn [{:keys [this]}]
      (this :load-agora-method))
    :load-agora-method
    (fn [{:keys [props state]} & [method-ref]]
-     (let [{:keys [namespace name snapshotId]} (or method-ref props)]
+     (let [{:keys [namespace name snapshotId]} (or method-ref props)
+           method (:method props)]
        (endpoints/call-ajax-orch
         {:endpoint (endpoints/get-agora-method namespace name snapshotId)
```

Comment:
not sure if this is the right place in the code ... but if the method snapshot is redacted, and snapshotId === -1, could you avoid making the ajax call at all? Making an API call for -1 will always fail, so we shouldn't even try.

Reply:
the snapshot id will never be -1. the index of the redacted method in the snapshot dropdown will be -1 for a redacted method, but this is not what is used when calling get-agora-method.